### ISSUE TITLE
UUID not a string pls

### ISSUE THREAD

**Issue by aDotInTheVoid** [gpuctl/gpuctl#129]

**Comment by danielg0** [comment]
I would like to propose this issue is closed.

The UUID can be converted to a 128-bit numeric value, but neither Go nor Postgres have a type for 128-bit ints, so we would have to use arbitrary precision integers. In [Postgres](https://www.postgresql.org/docs/16/datatype-numeric.html#DATATYPE-NUMERIC-DECIMAL), these are stored as BCDs and based on some quick calculations, I don't believe changing to them would give us much of a space/speed advantage.

It's true that it guarantees that the UUID is the 128-bit number we expect, but as we don't perform any numeric operations on the value, only equality checks, I don't see why a string rep isn't appropriate.

TLDR: I don't currently see a benefit to storing UUID as not a string that outweighs the complexity cost of parsing to/from the string representation

**Comment by aDotInTheVoid** [comment]
@NathanielB123 this is your issue, What Do You Think

CC @sirjacobofglais , i think this came up a lot for you last summer

**Comment by sirjacobofglais** [comment]
it's (extremely) badly abstracted as a string - using that datatype is just misunderstanding what a UUID is.

if representing it any other way would be significantly clunkly then I think it's fine to keep it like this, BUT it definitely needs a long comment on the typedef to explain why we are doing this. And we should keep this issue in mind if we run into stringy problems with this in the future.

**Comment by NathanielB123** [comment]
I guess the abstraction concern largely comes down to how numbers can be represented multiple ways by a string (decimal, hex, base64 etc...)? - if so, commenting/documentation sounds like a reasonable solution to me.

**Comment by danielg0** [comment]
> The UUID can be converted to a 128-bit numeric value, but neither Go nor Postgres have a type for 128-bit ints

My argument here is invalidated. Whilst it is true that Postgres doesn't have a type for 128-bit integers, it has a type for UUIDs (https://www.postgresql.org/docs/16/datatype-uuid.html) :facepalm:
